doc/intro.md

```diff
@@ -96,28 +124,37 @@ A **tuple** is multiple expressions joined with `,`, as in `foo bar > baz, zot`.
 
 This allows for open methods (see below) to be defined for arbitrary lengths of tuples.
 
-## Lambdas and patterns
+## Pattern-Matching
 
-It's expected that lambdas will be a common thing to type on the repl, so the syntax has been kept very lightweight.  A basic lambda looks like this:
+Things get a little more interesting when tagwords are combined with pattern-matching in lambdas.  Rather than merely binding parameter names to argument values, lambdas can also pattern-match and destructure tagged values over multiple clauses:
 
 ``` tulip
-[ foo => some-expr-in foo ]
+map f = [ .nil => .nil; .cons x xs => .cons (f x) (map f xs) ]
 ```
 
-where the variable `foo` is bound inside the block.  Nested blocks have the usual properties of closures.
+To accompany the implicit "or" between clauses, `:` can be used as an "and" within a clause:
```

Comment:
oh! yeah. there *is* a separate feature for disjunction of patterns, but it doesn't allow binding. it would be

```
@match %foo [ pat1; pat2 ]

thing (x : %foo) = ...
```

where it matches if *either* pat1 or pat2 apply.